File: reactive-web/src/main/java/com/github/corneil/reactivespring5/repository/LocationHistoryRepository.java
package com.github.corneil.reactivespring5.repository;

import com.github.corneil.model.LocationHistory;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Flux;

import java.util.Date;

public interface LocationHistoryRepository extends ReactiveCrudRepository<LocationHistory, String> {
	Flux<LocationHistory> findByTimestampBetween(Date startDate, Date endDate);
}

File: reactive-kotlin/src/main/kotlin/com/github/corneil/reactivespring5kotlin/repository/LocationHistoryRepository.kt
package com.github.corneil.reactivespring5kotlin.repository

import com.github.corneil.reactivespring5kotlin.model.LocationHistory
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux
import java.util.*

interface LocationHistoryRepository : ReactiveCrudRepository<LocationHistory, String> {
	fun findByTimestampBetween(startDate: Date, endDate: Date): Flux<LocationHistory>
}
